### PR TITLE
Create new view for importing

### PR DESCRIPTION
**Pull request by Aliciawyse**
### Issue
Fix for https://github.com/Terrastories/terrastories/issues/592

### Questions
- What should the headers be in the sample csv? Asking because when you view these pages in the app (`http://localhost:3000/admin/places?locale=en` and `http://localhost:3000/admin/speakers?locale=en`), you'll see a table with only a few aspects of places or speakers displayed.

- Is it a problem that I can successfully upload a csv for places with only name, description and region as the csv headers?

- This JS test keeps failing. I'm not sure how to confirm if this failure is caused by my changes. I think it's because I removed a `section` from the index/list admin page of speakers, places and stories
```
 FAIL  spec/javascript/components/Card.spec.js (11.449s)
  ● Card component › Given a user props › With an editor role › Renders the correct information

    expect(jest.fn()).toHaveBeenCalledTimes(2)

    Expected mock function to have been called two times, but it was called one time.

      28 | 
      29 |         expect(wrapper.find('.card--tasks').childAt(0)).toHaveLength(1);
    > 30 |         expect(global.I18n.currentLocale).toHaveBeenCalledTimes(2);
```

### CODE REVIEW

File: rails/app/views/admin/stories/import_page.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<section class="main-content__body padding-fix color-fix">
+  <p>
+    IMPORTANT: Need to import stories? The order of the values in your csv must be in a particular order. Feel free to use this <%= link_to "SAMPLE CSV", export_sample_csv_admin_stories_path %> to get started with successfully uploading stories.
+  </p>
+
+  <%= form_tag send("import_csv_admin_stories_path"), multipart: true do %>
```
Comment (Aliciawyse) [inline rails/app/views/admin/stories/import_page.html.erb:6]: Something feels weird about this change from `import_csv_admin_#{page.resource_name.pluralize}_path` to `import_csv_admin_stories_path`

Comment (FeminismIsAwesome) [reply rails/app/views/admin/stories/import_page.html.erb:6]: @Aliciawyse i think thats totally fine! We're moving from a generic form to forms that are more specific to the resource's need. I've found the administrate abstraction to usually get in the way anyway; anything thats more specific and rails-y feels a little easier to read too. 

If we do end up wanting to generecize it again, we could always find the common pieces between the forms and make a partial that goes back to the `import_csv_admin_#{page.resource_name.pluralize}_path` style.

Comment (lauramosher) [reply rails/app/views/admin/stories/import_page.html.erb:6]: @Aliciawyse I agree with @FeminismIsAwesome in this; the form tag itself is going to be "duplicated" across each specialized view. The Administrate "way" forced us to do weird things like interpolating the resource into the path and I for one am glad to see that go.

IF we wanted to continue to share some of the logic, we could create a partial view for the form; but I'm okay with waiting on this until we see the repetition.

File: rails/app/controllers/admin/speakers_controller.rb [hunk 2]
```diff
@@ -63,5 +63,13 @@ def default_sorting_attribute
     def default_sorting_direction
       :asc
     end
+
+    def import_page
+      render "import_page"
+    end
+
+    def export_sample_csv
+      send_data Speaker.export_sample_csv, filename: "sample_places.csv"
```
Comment (FeminismIsAwesome) [inline rails/app/controllers/admin/speakers_controller.rb:72]: @Aliciawyse i think you meant sample_speakers.csv! :D

File: rails/spec/models/speaker_spec.rb [hunk 3]
```diff
@@ -136,4 +136,10 @@
       it { is_expected.to be_nil }
     end
   end
+
+  describe 'export_sample_csv' do
+    it 'downloads a csv' do
+      expect(described_class.export_sample_csv).to eq("name,birthdate,community\n")
```
Comment (FeminismIsAwesome) [inline rails/spec/models/speaker_spec.rb:142]: speakers are missing the media column

File: rails/spec/models/place_spec.rb [hunk 4]
```diff
@@ -133,4 +133,10 @@
       expect(place.point_geojson.keys).to include('properties')
     end
   end
+
+  describe 'export_sample_csv' do
+    it 'downloads a csv' do
+      expect(described_class.export_sample_csv).to eq("name,description,region\n")
```
Comment (FeminismIsAwesome) [inline rails/spec/models/place_spec.rb:139]: places also have a lat, long, and media.

**Comment by FeminismIsAwesome** [comment]
@Aliciawyse some replies to your questions):

1) try adding the columns i suggested to the default csvs (it was missing a few, which is fair it was undocumented so hard to spot, i just looked at FileImport::PlaceRowDecorator and the Speaker code to see what was up when it failed to import my speakers). That hopefully should convert the nil issues to blanks, but even if they persist, I'd accept this code and then make a follow up feature to not bug out if media files are missing on speaker/story. You can see the bug on line 23 of the CsvHelper class.

2) that's a great question re: the js tests. I thought it was weird, so I checked myself by moving from your branch to master, which should have none of your changes, and seeing the same test fail. So I think we just accidentally merged in a failing test, and I'll add a ticket to fix that.

Let me know if you have further questions!

File: rails/app/controllers/admin/places_controller.rb [hunk 5]
```diff
@@ -52,9 +52,17 @@ def import_csv
...
         filepath = params[:file].read
         errors = Place.import_csv(filepath, current_community)
-        errors.empty? ? flash[:notice] = "Places were imported successfully!" : flash[:error] = errors
+        errors.empty? ? flash[:notice] = "Places were imported successfully! View them #{view_context.link_to 'here', admin_places_path}." : flash[:error] = errors
         redirect_back(fallback_location: root_path)
       end
     end
+
+    def import_page
+      render "import_page"
```
Comment (FeminismIsAwesome) [inline rails/app/controllers/admin/places_controller.rb:61]: @Aliciawyse are those renders necessary? doesn't rails try to render a page with matching name by default?

**Review by FeminismIsAwesome — APPROVED** [review]
had one more question re: a render call, but honestly lgtm

**Comment by Aliciawyse** [comment]
oh good question. I'll double check that and follow up

**Pull request merged**